File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> v;
        for (int i = 0; i < nums1.size(); i++) {
            for (int j = 0; j < nums2.size(); j++) {
                if (nums1[i] == nums2[j]) {
                    nums2[j] = -1;
                    v.push_back(nums1[i]);
                }
            }
        }

        // sort + unique + erase method
        sort(v.begin(), v.end());
        auto it = unique(v.begin(), v.end());
        v.erase(it, v.end());
        return v;
    }
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set1 = new HashSet<>();
        HashSet<Integer> ans = new HashSet<>();
        for (int i=0; i<nums1.length; i++){
            set1.add(nums1[i]);
        }
        
        for (int i=0; i<nums2.length; i++){
            if (set1.contains(nums2[i])){
                ans.add(nums2[i]);
            }
        }

        int[] array = new int[ans.size()];
        int k = 0;
        for (int i: ans){
            array[k++] = i;
        }

        return array;
    }
}